File: js/test/test_solution_1.mjs
import solution_1 from "../src/solution_1.mjs"

export default class Test {
  static _contadorTestesFalharam = 0
  static _contadorTestesPassaram = 0
  constructor() {}

  static test(textoEsperado, textoRecebido) {
    Test._contadorTestesFeitos++
    if (textoEsperado === textoRecebido) {
      Test._contadorTestesPassaram++
      //console.log(`Teste ${Test._contadorTestesFeitos} Passou!`)
    } else {
      Test._contadorTestesFalharam++
      let testAtual =
        Test._contadorTestesFalharam + Test._contadorTestesPassaram
      console.error("\n--------------------------------------------------")
      console.error(`Teste ${testAtual} Falhou!`)
      console.error(`O texto esperado era: ${textoEsperado}`)
      console.error(`O texto recebido foi: ${textoRecebido}`)
      console.error("--------------------------------------------------\n")
    }
  }
  static relatorio() {
    console.error("\n\n--------------------------------------------------")
    console.log("Relatório de Testes:")
    console.log(
      `Testes Feitos: ${
        Test._contadorTestesPassaram + Test._contadorTestesFalharam
      }`
    )
    console.log(`Testes Passaram: ${Test._contadorTestesPassaram}`)
    console.error(`Testes Falharam: ${Test._contadorTestesFalharam}`)
    console.error("--------------------------------------------------\n\n")
  }
}

Test.test(
  "O mago atacou usando magia",
  new solution_1("Gandalf", 1000, "mago").atacar()
)
Test.test(
  "O guerreiro atacou usando espada",
  new solution_1("Aragorn", 25, "guerreiro").atacar()
)
Test.test(
  "O monge atacou usando artes marciais",
  new solution_1("Bruce Lee", 30, "monge").atacar()
)
Test.test(
  "O ninja atacou usando shuriken",
  new solution_1("Naruto", 16, "ninja").atacar()
)
Test.relatorio()

File: python/test/test_solution_1.py
from Test import Test
import sys
import os
# Adiciona a pasta "src" ao caminho do sistema
src_path = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'src'))
sys.path.append(src_path)

from module.Hero import Hero




Test.assert_equals("O mago atacou usando magia", Hero('Gandalf', 1000, 'mago').attack(),"1º - testa se a instancia da class Hero responde corretamente quando executado o método attack")
Test.assert_equals("O guerreiro atacou usando espada", Hero("Kenshin Himura", 28, 'guerreiro').attack(), "2º - testa se a instancia da class Hero responde corretamente quando executado o método attack")
Test.assert_equals("O monge atacou usando artes marciais", Hero('Tenzin', 51, 'monge').attack(), "3º - testa se a instancia da class Hero responde corretamente quando executado o método attack")
Test.assert_equals("O ninja atacou usando shuriken", Hero('Kakashi', 30, 'ninja').attack(), "4º - testa se a instancia da class Hero responde corretamente quando executado o método attack")
Test.print_summary()